### PR TITLE
Update algorithms for Python3 - Group 4

### PR DESCRIPTION
**Pull request by rosswhitfield**
This is Group 4 of #16739 

**To test:**
- Code review
- Make sure the unittests still pass with python2.
- Test that the related unittests pass when using python3.

Does not need to be in the release notes.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [x] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by peterfpeterson** [comment]
After merging this, tests that fail in python2 are:
```
$ ctest -R Python
...
469 - PythonInterfaceAPI_MultipleFilePropertyTest (Failed)
539 - PythonAlgorithms_LoadLogPropertyTableTest (Failed)
```
which are both broken on my system with nexus 4.4.1 anyhow. See #17001.

Lots more fail in python3. However, none are relevant to this PR.

I also reviewed the changes (very unexciting) and it is fine.

:shipit:

File: Framework/PythonInterface/plugins/algorithms/LRAutoReduction.py [hunk 1]
```diff
@@ -425,7 +426,7 @@ def _get_direct_beam(self, meta_data_run, data_set):
         direct_beam_runs_str = self._read_property(meta_data_run, "direct_beam_runs",
                                                    _direct_beam_runs, is_string=True)
         # The direct runs in the DAS logs are stored as a string
-        if isinstance(direct_beam_runs_str, (str, unicode)):
+        if isinstance(direct_beam_runs_str, str):
```
Comment (MichaelWedel) [inline Framework/PythonInterface/plugins/algorithms/LRAutoReduction.py]: Will this still do the right thing in Python 2 for unicode strings?

Comment (rosswhitfield) [inline Framework/PythonInterface/plugins/algorithms/LRAutoReduction.py]: No it doesn't. `isinstance(u"foobar",(str,unicode)) = True` while `isinstance(u"foobar",str) = False`. But `unicode` isn't a type in python 3 so something else is needed here.

How about `isinstance(u"foobar",(str,type(u'')))`? It works for python 2 and 3. Unless there is a better way?

Comment (MichaelWedel) [inline Framework/PythonInterface/plugins/algorithms/LRAutoReduction.py]: I just found this:

```
from six import string_types

if isinstance(direct_beam_runs_str, string_types):
    ....
```
But haven't tested it. From the docs:

"six.string_types: Possible types for text data. This is basestring() in Python 2 and str in Python 3."

Comment (rosswhitfield) [inline Framework/PythonInterface/plugins/algorithms/LRAutoReduction.py]: Yeah that works. Thanks!

**Pull request merged**